azurelinuxagent/common/protocol/wire.py

```diff
@@ -675,14 +672,14 @@ def _fetch_response(self, uri, headers=None, use_proxy=None):
                                                     is_healthy=not restutil.request_failed_at_hostplugin(resp),
                                                     source='WireClient',
                                                     response=error_response)
-                raise ProtocolError(msg)
+                raise HttpResponseFailure(msg)
             else:
                 if host_plugin is not None:
                     host_plugin.report_fetch_health(uri, source='WireClient')
 
         except (HttpError, ProtocolError, IOError) as e: # pylint: disable=C0103
             logger.verbose("Fetch failed from [{0}]: {1}", uri, e)
-            if isinstance(e, ResourceGoneError) or isinstance(e, InvalidContainerError): # pylint: disable=R1701
+            if isinstance(e, ResourceGoneError) or isinstance(e, InvalidContainerError) or isinstance(e, HttpResponseFailure): # pylint: disable=R1701
```

Comment:
Since we're changing this line anyway, it might be worth it to fix the `consider-merging-isinstance<R1701>` linter error here.
The easiest fix would basically replace this line with:

```python
if isinstance(e, (ResourceGoneError, InvalidContainerError, HttpResponseFailure)):
```

Alternatively, we could more selectively `except` on these errors specifically, and then have a more general `except` after.